packages/server/worker/src/lib/consume/worker-job-rate-limiter.ts

```diff
@@ -1,14 +1,38 @@
 import { apDayjsDuration, getPlatformPlanNameKey, getProjectMaxConcurrentJobsKey } from '@activepieces/server-shared'
-import { ApEdition, ExecuteFlowJobData, isNil, JobData, PlanName, PlatformId, ProjectId, RunEnvironment, WorkerJobType } from '@activepieces/shared'
+import {
+    ApEdition,
+    assertNotNullOrUndefined,
+    ExecuteFlowJobData,
+    isNil,
+    JobData,
+    PlanName,
+    PlatformId,
+    ProjectId,
+    RunEnvironment,
+    WorkerJobType
+} from '@activepieces/shared'
 import { FastifyBaseLogger } from 'fastify'
 import { workerMachine } from '../utils/machine'
 import { workerDistributedStore, workerRedisConnections } from '../utils/worker-redis'
+import { jobQueue } from '../../../../api/src/app/workers/queue/job-queue';
+import { JobType } from '../../../../api/src/app/workers/queue/queue-manager';
+import { throttledJobQueue } from '../../../../api/src/app/workers/queue/throttled-job-queue';
 
 
 export const RATE_LIMIT_WORKER_JOB_TYPES = [WorkerJobType.EXECUTE_FLOW]
-const projectSetKey = (projectId: string): string => `active_jobs_set:${projectId}`
+const projectActiveSetKey = (projectId: string): string => `active_jobs_set:${projectId}`
+const projectWaitingSetKey = (projectId: string): string => `waiting_jobs_set:${projectId}`
 
 export const workerJobRateLimiter = (_log: FastifyBaseLogger) => ({
+    async throttleJob(id: string, data: JobData): Promise<void> {
+        const redis = await workerRedisConnections.useExisting()
+        const key = projectWaitingSetKey(data.projectId!)
+
+        await redis.zadd(key, Date.now(), id)
```

Comment:
If we are going by the time, It's better to use queue as It's O(1) fore pop, but I would suggest we go by the priority there and time or plain queue.

You choose.

https://redis.io/glossary/redis-queue/